services/app-web/src/pages/CMSDashboard/CMSDashboard.tsx

```diff
@@ -110,39 +101,64 @@ export const CMSDashboard = (): React.ReactElement => {
                 // should not display draft submissions to a CMS user - this is also filtered out on the api side
                 return
             }
+
+            // Calculate proper submission data to display in each row
+            // if UNLOCKED package, form data could be in progress of being edited, show the static form data from the last submitted revision
+            // also, use lastUpdated from unlockData (if unlocked) or last submit (if initial submission)
+            // otherwise, use data from current revision for Submitted or Resubmitted packages
+            let packageDataToDisplay = currentFormData
+            let lastUpdated = currentFormData.updatedAt
+            if (sub.status === 'UNLOCKED') {
+                const previousRevision = sub.revisions[1]
+                const previousFormData = base64ToDomain(
+                    previousRevision.node.formDataProto
+                )
+
+                if (previousFormData instanceof Error) {
+                    recordJSException(
+                        `indexHealthPlanPackagesQuery: Error decoding proto for display of an unlocked submission. ID: ${sub.id} Error message: ${previousFormData.message}`
+                    )
+
+                    return
+                }
+
+                packageDataToDisplay = previousFormData
+
+                if (
+                    !previousRevision?.node?.unlockInfo?.updatedAt &&
+                    !previousRevision?.node?.submitInfo?.updatedAt
+                ) {
+                    recordJSException(
+                        `indexHealthPlanPackagesQuery: Error finding updatedAt for an UNLOCKED SUBMISSION. Check the revision data is properly formatted. ID: ${sub.id}`
+                    )
+
+                    return
+                }
+                lastUpdated =
```

Comment:
TODO: add a comment here.Last updated at value should be CMS unlock time (if its unlocked) or else state submit time (if its just submitted), fall back to last updated at  for the current revision (likely we wouldn't need that but just to have something there).